### ISSUE TITLE
[ENH] - <title> Open To Colab

### ISSUE THREAD

**Issue by NicolasChaillou** [jupyter/docker-stacks#1680]
### What docker image(s) is this feature applicable to?

base-notebook

### What changes are you proposing?

1 : Hello can you add in `/etc/jupyter/jupyter_notebook_config.py`
the line : `c.NotebookApp.allow_origin = 'https://colab.research.google.com/'`

2: And set a path to /vagrant folder in built-in?

That's would help

Thanks

### How does this affect the user?

1: Help to use Google Colab in native way

2: helps to populate in VM with a share folder

### Anything else?

All my regards, and thank you for all

**Comment by minrk** [comment]
This would mean allowing cross-origin requests from any notebook on google colab for all notebook servers by default. I don't think that's an appropriate default. I suspect perhaps the meaning of `allow_origin` is misunderstood?

Note: this does _not_ affect using these images on any domain, which already works. Only _cross-origin_ requests are affected by this (i.e. requests to a server on one host from a page served by a _different_ host). Can you describe what you want to accomplish in more detail?

> And set a path to /vagrant folder in built-in?

I'm not sure what this would help, since this is a collection of docker builds, not VM configurations. What is this for?

**Comment by NicolasChaillou** [comment]
Hello Min, my requests were effectively to manage the docker inside a VM and by Google Colab notebooks.

I know it’s not base on the container features appropriately, even in the start of the build of these images.

I just had troubles by launching automatically these 2 features at the docker run /start, meaning I thought it might be set Inside the the container itself.

So if you have a link to share on how to do it afterward, I’ll be please.

Best Regards

Nico

Envoyé à partir de Courrier<https://go.microsoft.com/fwlink/?LinkId=550986> pour Windows

De : Min ***@***.***>
Envoyé le :dimanche 10 avril 2022 20:44
À : ***@***.***>
Cc : Nicolas ***@***.***>; ***@***.***>
Objet :Re: [jupyter/docker-stacks] [ENH] - <title> Open To Colab (Issue #1680)


This would mean allowing cross-origin requests from any notebook on google colab for all notebook servers by default. I don't think that's an appropriate default. I suspect perhaps the meaning of allow_origin is misunderstood?

Note: this does not affect using these images on any domain, which already works. Only cross-origin requests are affected by this (i.e. requests to a server on one host from a page served by a different host). Can you describe what you want to accomplish in more detail?

And set a path to /vagrant folder in built-in?

I'm not sure what this would help, since this is a collection of docker builds, not VM configurations. What is this for?

**Comment by mathbunnyru** [comment]
@NicolasChaillou could you please write what you try to accomplish, how you do it, and what doesn't work (and you think it should).

As @minrk pointed out, `c.NotebookApp.allow_origin = 'https://colab.research.google.com/'` is not a proper default.
If you really want to have this, you can easily achieve this by basing your image on one of our images.

**Comment by mathbunnyru** [comment]
Closing this one, because no reply was received.

**Comment by iaSalazar** [comment]
I want to host the spark image in one of my GCP instances but i cannot upload or create folders in it when I access Jupyter from my personal pc, I came across this thread.

I tried modifying the config.py file with:

c.NotebookApp.allow_origin = '*'
c.NotebookApp.ip = '0.0.0.0'

but the issue persist.

Any ideas how can achieve that? @mathbunnyru I did not wanted to open a new issue but i think it is related to what @NicolasChaillou did.
